Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/threshold/BasicThresholder.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.parkservices.threshold;
+
+import static com.amazon.randomcutforest.CommonUtils.checkArgument;
+
+import java.util.List;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class BasicThresholder implements IThresholder {
+
+    // a parameter to make high score regions a contiguous region as opposed to
+    // collection of points in and out of the region
+    protected double elasticity = 0.01;
+
+    // keeping a count of the values seen because both variables may not be used
+    protected int count = 0;
+
+    // horizon = 0 is short term, switches to secondary
+    // horizon = 1 long term, switches to primary
+    protected double horizon = 0.5;
+
+    // below these many observations, deviation is not useful
+    protected int minimumScores = 10;
+
+    protected Deviation primaryDeviation;
+
+    protected Deviation secondaryDeviation;
+
+    // fraction of the grade that comes from absolute scores in the long run
+    protected double absoluteScoreFraction = 0.5;
```

Comment:
This is a leftover from previous version -- the idea is that if we want some fraction of the anomaly grade to come from the absolute scores. It may be that we do not use it -- we will delete it at that point.